src/nvim/msgpack_rpc/channel.c

```diff
@@ -823,6 +823,7 @@ static void log_server_msg(uint64_t channel_id,
   msgpack_unpack_next(&unpacked, packed->data, packed->size, NULL);
   uint64_t type = unpacked.data.via.array.ptr[0].via.u64;
   DLOGN("[msgpack-rpc] nvim -> client(%" PRIu64 ") ", channel_id);
+  log_lock();
```

Comment:
What I meant is that specialized log functions like these should be rare enough to ignore this single case where the log mutex is managed outside the log module. I'd say at least 90% of the cases we'll just use the standard log macros which are already synchronized.

If we start to see this pattern more, then it might be worth refactoring.